llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -4010,6 +4094,12 @@ void AArch64TTIImpl::getUnrollingPreferences(Loop *L, ScalarEvolution &SE,
       EnableFalkorHWPFUnrollFix)
     getFalkorUnrollingPreferences(L, SE, UP);
 
+  if (ST->getProcFamily() == AArch64Subtarget::AppleA14 ||
+      ST->getProcFamily() == AArch64Subtarget::AppleA15 ||
+      ST->getProcFamily() == AArch64Subtarget::AppleA16 ||
+      ST->getProcFamily() == AArch64Subtarget::AppleM4)
+    getAppleRuntimeUnrollPreferences(L, SE, UP, *this);
```

Comment:
Mind doing a switch instead? Each time we add one, it'll be a one-line addition, instead of a `-+`, and it's a bit more friendly to having non-upstream ones.

Reply:
Done thanks! Also included the code for Falkor from above in the new switch